lib/ruby_lsp/ruby_lsp_rails/addon.rb

```diff
@@ -35,7 +35,16 @@ def initialize
         Thread.new do
           @addon_mutex.synchronize do
             # We need to ensure the Rails client is fully loaded before we activate the server addons
-            @client_mutex.synchronize { @rails_runner_client = RunnerClient.create_client }
+            @client_mutex.synchronize do
```

Comment:
We intentionally don't want to slow down booting the main language server, which is why the client mutex is not locked here.

With the current mechanism, anyone add-on invoking `rails_runner_client` will be stuck until booting the runner client finishes, but it will not block indexing by the main LSP.

We tested this in the Tapioca add-on we're building and this code guarantees correct ordering without blocking the main thread:

```ruby
def activate(global_state, outgoing_queue)
  # get a handle for the addon
  addon = ::RubyLsp::Addon.get("Ruby LSP Rails")

  Thread.new do
    # get a handle to the runner client. This will block this new thread until
    # the Rails add-on finishes initializing and unlocks the client mutex, but
    # the main thread will continue to run properly without stopping indexing
    @rails_runner_client = addon.rails_runner_client
          
    @rails_runner_client.register_server_addon(File.expand_path("server_addon.rb", __dir__))
end
```

You can check the rest of the approach in https://github.com/Shopify/tapioca/pull/2001.

Reply:
Ah ok, fair enough! I wasn't aware of `::RubyLsp::Addon.get`, but it should certainly make things a bit easier